docs/source/model_doc/m2m_100.rst

```diff
@@ -0,0 +1,125 @@
+to effectively increase model capacity through a combination of dense scaling and language-specific sparse parameters
+to create high quality models. Our focus on non-English-Centric models brings gains of more than 10 BLEU when directly
+translating between non-English directions while performing competitively to the best single systems of WMT. We
+open-source our scripts so that others may reproduce the data, evaluation, and final M2M-100 model.*
+
+
+Training and Generation
+_______________________________________________________________________________________________________________________
+
+M2M100 is a multilingual encoder-decoder (seq-to-seq) model primarily intended for translation tasks. As the model is
+multilingual it expects the sequences in a certain format: A special language id token is used as prefix in both the
+source and target text. The source text format is :obj:`[lang_code] X [eos]`, where :obj:`lang_code` is source language
+id for source text and target language id for target text, with :obj:`X` being the source or target text.
+
+- Supervised Training
+
+.. code-block::
+
+    from transformers import M2M100Config, M2M100ForConditionalGeneration, M2M100Tokenizer
+
+    model = M2M100ForConditionalGeneration.from_pretrained('facebook/m2m100_418M')
+    tokenizer = M2M100Tokenizer.from_pretrained('facebook/m2m100_418M', src_lang="en", tgt_lang="fr")
+
+    src_text = "Life is like a box of chocolates."
+    tgt_lang = "La vie est comme une boîte de chocolat."
+
+    model_inputs = tokenizer(src_text, return_tensors="pt")
+    with tokenizer.as_target_tokenizer():
+        labels = tokenizer(tgt_text, return_tensors="pt").input_ids
+
+    loss = model(**model_inputs, labels=labels) # forward pass
+
+
+- Generation
+
+    M2M100 uses the :obj:`eos_token_id` as the :obj:`decoder_start_token_id` for generation with the target language id
+    being forced as the first generated token. To force the target language id as the first generated token, pass the
+    `forced_bos_token_id` parameter to the `generate` method. The following example shows how to translate between
+    Hindi to French and Chinese to English using the `facebook/m2m100_418M` checkpoint.
+
+.. code-block::
+
+    from transformers import M2M100ForConditionalGeneration, M2M100Tokenizer
+
+    hi_text = "जीवन एक चॉकलेट बॉक्स की तरह है।"
+    chinese_text = "生活就像一盒巧克力。"
+
+    model = M2M100ForConditionalGeneration.from_pretrained("facebook/m2m100_418M")
+    tokenizer = M2M100Tokenizer.from_pretrained("facebook/m2m100_418M")
+
+    # translate Hindi to French
+    tokenizer.src_lang = "hi"
+    encoded_hi = tokenizer(hi_text, return_tensors="pt")
+    generated_tokens = model.generate(**encoded_hi, forced_bos_token_id=tokenizer.get_lang_id("fr"))
+    tokenizer.batch_decode(generated_tokens, skip_special_tokens=True)
+    # => "La vie est comme une boîte de chocolat."
```

Comment:
Since there is a result here, this should use the doctest syntax (so >>> for each line of code, nothing for the return.